Cleanup weather type doc (#45249)
CleverRaven/Cataclysm-DDA: doc/WEATHER_TYPE.md

| `rains`                        | Whether said precipitation falls as rain.                             |
| `acidic`                       | Whether said precipitation is acidic.                                 |
| `tiles_animation`              | Optional, name of the tiles animation to use                          |

> | `sound_category`               | Optional, if playing sound effects what to use. Valid values are: silent, drizzle, rainy, thunder, flurries, snowstorm and snow. |

Reviewer: A bit unclear what is meant by "if playing sound effects what to use". "what sound effect to use by the sound system"?

Reviewer 2: "what sound effect to play"?

> | `sound_category`               | Optional, if playing sound effects what to use. Valid values are: silent, drizzle, rainy, thunder, flurries, snowstorm and snow. |

Reviewer 2: ```suggestion
| `sound_category`               | Optional, what sound effect to play. Valid values are: silent, drizzle, rainy, thunder, flurries, snowstorm and snow. |
```

| `sun_intensity`                | Strength of the sun. Valid values are: none, light, normal, and high  |
| `duration_min`                 | Optional, the lower bound on the amount of time this weather can last. Defaults to 5 minutes. Unless time_between_min and time_between_max are set the weather can happen again as soon as it ends. |
| `duration_max`                 | Optional, the upper bound on the amount of time this weather can last. Defaults to 5 minutes. Unless time_between_min and time_between_max are set the weather can happen again soon as it ends. |

> | `time_between_min`             | Optional, the lower bounds of the amount of time that will be guranteed to pass before this weather happens again. Defaults to 0. |
> | `time_between_max`             | Optional, the upper bounds of the amount of time that will be guranteed to pass before this weather happens again. Defaults to 0. |

Reviewer 2: ```suggestion
| `time_between_min`             | Optional: the lower bound on the amount of time that will be guaranteed to pass before this weather happens again. Defaults to 0. |
| `time_between_max`             | Optional: the upper bound on the amount of time that will be guaranteed to pass before this weather happens again. Defaults to 0. |
```

| `weather_animation`            | Optional, Information controlling weather animations.  Members: factor, color and glyph |
| `effects`                      | Array for the effects the weather has. Descibed in detail below

> | `requirements`                 | Optional, is what determines what weather it is.  All members are optional. When checking what weather it is it loops through the entries in order and uses the last one to succeed. |

Reviewer: How about something like

> Optional, a list of conditions that have to be met for the weather to appear.

Reviewer 2: And does "succeed" include just on the requirements, or on the chance of a change to that weather  type from the current weather?

    "precip": "heavy",
    "rains": true,
    "acidic": false,
    "effects": [
      {
        "one_in_chance": 50,
        "must_be_outside": false,
        "sound_message": "You hear a distant rumble of thunder.",
        "sound_effect": "thunder_far"
      },
      {
        "one_in_chance": 600,
        "must_be_outside": false,
        "message": "A flash of lightning illuminates your surroundings!.",
        "sound_effect": "thunder_near",
        "lightning": true
      }
    ],
    "tiles_animation": "weather_rain_drop",
    "weather_animation": { "factor": 0.04, "color": "light_blue", "sym": "," },
    "sound_category": "thunder",
    "sun_intensity": "none",
    "requirements": { "pressure_max": 990, "required_weathers": [ "thunder" ] }
  },
]
```

> ### `effects` properties

Reviewer 2: ```suggestion
### Weather effect properties
```

      ],
      "spawns":
      [{
        "max_radius": 10,
        "min_radius": 10,
        "target": "mon_zombie_survivor_elite",
        "hallucination_count": 1,
        "real_count": 0
      }]
}
```

### `requirements` properties

|     Identifier                 |                              Description                              |
| ------------------------------ | --------------------------------------------------------------------- |

> | `pressure_min`                 | |

Reviewer 2: ```suggestion
| `pressure_min`                 | These are all minimum and maximum values for which the weather will occur.  I.e., it will only rain if it is sufficiently humid. |
```

| `pressure_max`                 | |
| `humidity_min`                 | |
| `humidity_max`                 | |
| `temperature_min`              | |
| `temperature_max`              | |
| `windpower_min`                | |

> | `windpower_max`                | These are all minimum and maximum values for which the weather will occur.  I.e it will only rain if its humid enough |

Reviewer 2: ```suggestion
| `windpower_max`                |  |
```

> | `humidity_and_pressure`        | if there are pressure and humidity requirements are they both required or just one |

Reviewer: ```suggestion
| `humidity_and_pressure`        | should logical AND be used for pressure and humidity requirements when they both are defined |
```

| `acidic`                       | does this require acidic precipitation                                |

> | `time`                         | Valid values are: day, night, and both.                               |

Reviewer: ```suggestion
| `time`                         | Valid values are: "day", "night", and "both".                               |
```
avoiding ambiguity

> | `required_weathers`            | a string array of possible weathers it is at this point in the loop. i.e. rain can only happen if the conditions for clouds light drizzle or drizzle are present |

Reviewer: ```suggestion
| `required_weathers`            | a string array of possible weathers that precede this one. i.e. rain can only happen if the conditions for clouds light drizzle or drizzle are present |
```
Provided "string array" is equivalent to "an array of strings". Not sure "precede" is the wording either.

> | `required_weathers`            | a string array of possible weathers it is at this point in the loop. i.e. rain can only happen if the conditions for clouds light drizzle or drizzle are present |

Reviewer 2: ```suggestion
| `required_weathers`            | a string array of possible prior weathers; i.e., rain can only happen if the conditions for clouds, light drizzle, or drizzle are present |
```
Is it that the conditions (temperature/humidity/etc) for those weathers are met, or that the weathers were actually present?

> | `required_weathers`            | a string array of possible weathers that precede this one. i.e. rain can only happen if the conditions for clouds light drizzle or drizzle are present |

Author: ```suggestion
| `required_weathers`            | a string array of possible prior weathers; i.e., rain can only happen if the conditions for clouds, light drizzle, or drizzle are present |
```

> | `time_passed_min`              | Optional: Until this much time post cataclysm this weather won't happen.|

Reviewer: ```suggestion
| `time_passed_min`              | Optional: minimum time required to pass since cataclysm for this weather to appear.|
```

> | `time_passed_min`              | Optional: Until this much time post cataclysm this weather won't happen.|

Reviewer 2: ```suggestion
| `time_passed_min`              | Optional: Time after the Cataclysm when this weather can start appearing; |
```

> | `time_passed_min`              | Optional: minimum time required to pass since cataclysm for this weather to appear.|

Author: ```suggestion
| `time_passed_min`              | Optional: Time after the Cataclysm when this weather can start appearing; |
```

> | `time_passed_max`              | Optional: After this much time post cataclysm this weather stops.|

Reviewer 2: ```suggestion
| `time_passed_max`              | Optional: Time after the Cataclysm when this weather can no longer appear. |
```
